src/dmd/root/file.d

```diff
@@ -22,6 +22,379 @@ import dmd.root.filename;
+                }
+                if (handle != invalidHandle && .close(handle) != 0)
+                {
+                    error(null, 0, 0, "FileMapping: Cannot close file.\n");
+                    perror("close");
+                    fatal();
+                }
+            }
+            else version(Windows)
+            {
+                if (data.ptr !is null && UnmapViewOfFile(cast(void*) data.ptr) == 0)
+                {
+                    error(null, 0, 0, "FileMapping: Cannot unmap view of file.\n");
+                    fatal();
+                }
+                if (fileMappingObject != invalidHandle && CloseHandle(fileMappingObject) == 0)
+                {
+                    error(null, 0, 0, "FileMapping: Cannot close file mapping object.\n");
+                    fatal();
+                }
+                if (handle != invalidHandle && CloseHandle(handle) == 0)
+                {
+                    error(null, 0, 0, "FileMapping: Cannot close file.\n");
+                    fatal();
+                }
+            }
+            else static assert(0);
+        });
+    }
+
+    /**
+    Returns the zero-terminated file name associated with the mapping. Can
+    be saved beyond the lifetime of `this`.
+    */
+    const(char)* filename() const pure @nogc @safe nothrow { return name; }
+
+    /**
+    Frees resources associated with this mapping. However, it does not deallocate the name.
+    Reinitializes `this` as a fresh object that can be reused.
+    */
+    void close()
+    {
+        __dtor();
+        handle = invalidHandle;
+        data = null;
+        name = null;
+        version(Windows) fileMappingObject = invalidHandle;
+    }
+
+    /**
+    Deletes the underlying file and frees all resources associated. However, it
+    does not deallocate the name. Reinitializes `this` as a fresh object that
+    can be reused.
+
+    Returns: `true` iff the file was successfully deleted.
+    */
+    static if (!is(Datum == const))
+    bool discard()
+    {
+        // Truncate file to zero so unflushed buffers are not flushed unnecessarily.
+        resize(0);
+        auto deleteme = name;
+        close();
+        // In-memory resource freed, now get rid of the underlying temp file.
+        version(Posix)
+        {
+            import core.sys.posix.unistd;
+            if (unlink(deleteme) != 0)
+            {
+                perror("\tunlink error");
+                return false;
+            }
+        }
+        else version(Windows)
+        {
+            import core.sys.windows.winbase;
+            if (DeleteFileA(deleteme) == 0)
+            {
+                // TODO: add a perror equivalent?
+                fprintf(stderr, "\tDeleteFileA error %d\n", GetLastError());
+                return false;
+            }
+        }
+        else static assert(0);
+        return true;
+    }
+
+    /**
+    Returns `true` iff there is an active mapping associated with `this`.
```

Comment:
```suggestion
    Returns `true` if there is an active mapping associated with `this`.
```

Reply:
Stet, it means ["if and only if"](https://www.merriam-webster.com/dictionary/iff)